### ISSUE TITLE
Vectorial files and interactive plots

### ISSUE THREAD

**Issue by impact27** [spyder-ide/spyder#12211]
The plot panel is useful but I have a few problems with it:
 - The images are saved by default as low quality `png`. I would prefer `svg` and therefore need to:
```
from IPython.display import set_matplotlib_formats
set_matplotlib_formats('svg')
```
before plotting. I think it would be nice to have a setting in the plot widget to choose one of `'png', 'retina', 'jpeg', 'svg', 'pdf'`.

- The images are not shown if the backend is interactive (For example qt5). I think it would be great if there was a button to change the backend between inline and interactive, and if the interactive  backend plots where shown in the widget. This works by:

```
%matplotlib qt5
from IPython.core.display import display
import matplotlib.pyplot as plt
fig = plt.figure()
ax = fig.add_subplot()
ax.plot([1, 2, 4])
display(fig)
```
But this also displays the plot inline. One implementation could patch the figure `show` method and get the image data by:
```
%matplotlib qt5
import matplotlib.pyplot as plt
fig = plt.figure()
ax = fig.add_subplot()
ax.plot([1, 2, 4])
format = get_ipython().kernel.shell.display_formatter.format
format_dict, md_dict = format(fig)
```
and then send `format_dict` by the comm channel.

**Comment by goanpeca** [comment]
@jnsebgosselin ?

**Comment by ccordoba12** [comment]
> I think it would be nice to have a setting in the plot widget to choose one of 'png', 'retina', 'jpeg', 'svg', 'pdf'

There's one already:

    Preferences > IPython console > Graphics > Inline backend > Format

but it only covers png and svg. Retina is basically png with twice the size and @dalthviz is considering a good UI to include it at the moment.

Adding an option for pdf doesn't make much sense because we don't have a way to render pdfs using Qt only. But you could add an entry for jpeg there, if you want.

Also, to display SVG's we're using QtSvg. But that's not so good because Qt doesn't support the entire svg spec. Instead, we could use in Plots a QWebView and use it to better render svg's. You could also work on that, if you want.

> The images are not shown if the backend is interactive (For example qt5).

This is simply not possible at the moment because we can't embed an interactive Matplotlib backend on Plots. Besides, I have no idea how to capture the content that comes from a call to an interactive Matplotlib backend and redirect it to a Qt widget.

If you plan to work on that, please address it for Spyder 5 though.

**Comment by jnsebgosselin** [comment]
> This is simply not possible at the moment because we can't embed an interactive Matplotlib backend on Plots. Besides, I have no idea how to capture the content that comes from a call to an interactive Matplotlib backend and redirect it to a Qt widget.

I do not think it is what @impact27  is suggesting here. I think he is suggesting that even when in interactive mode, the plots should be added to the plots plugin, either in a svg or bitmap format, when show is called on the `Figure`.

I think this would be a great addition if we could add this functionality.

**Comment by jnsebgosselin** [comment]
So like something like the animation below. That way, we have an interactive figure on an independent window as well as a static image added to the plots plugin, which is very nice and exactly what I want most of the time! I would love to see this implemented.

![demo_plots](https://user-images.githubusercontent.com/10170372/78707481-ab494500-78de-11ea-89c3-f0fbc343d2a6.gif)

I know this is a topic that is very popular, but I do not see how embedding interactive matplotlib figures in Spyder is a good idea. The figures are going to look horrible most of the time because of size scaling...

**Comment by jnsebgosselin** [comment]
> I think it would be great if there was a button to change the backend between inline and interactive.

That also would be very cool.

**Comment by impact27** [comment]
> > I think it would be great if there was a button to change the backend between inline and interactive.
> 
> That also would be very cool.
> 
> If we could make it work without requiring a restart of the console, that would be even cooler.

We can just send `%matplotlib qt5` and `%matplotlib inline` messages to the console

**Comment by impact27** [comment]
> Adding an option for pdf doesn't make much sense because we don't have a way to render pdfs using Qt only. But you could add an entry for jpeg there, if you want.
> 
> Also, to display SVG's we're using QtSvg. But that's not so good because Qt doesn't support the entire svg spec. Instead, we could use in Plots a QWebView and use it to better render svg's. You could also work on that, if you want.

I think this would be solved nicely by:

```
from IPython.display import set_matplotlib_formats
set_matplotlib_formats('pdf', 'png')

import matplotlib.pyplot as plt
fig = plt.figure()
ax = fig.add_subplot()
ax.plot([1, 2, 4])
format = get_ipython().kernel.shell.display_formatter.format
format_dict, md_dict = format(fig)
```
Format dict contains a png that you can show in the widget and a pdf that you can save somewhere. Same for svg, you don't need Qt to read the svg as you could simply send both the svg (for saving) and the png (For showing).

**Comment by jnsebgosselin** [comment]
Yes, we should do this, this is a good idea.

**Comment by impact27** [comment]
I can do the communication with the kernel:
![plot](https://user-images.githubusercontent.com/6740194/78724644-33a30680-792e-11ea-9568-8d6d26b49496.gif)

**Comment by impact27** [comment]
![plot](https://user-images.githubusercontent.com/6740194/78755048-ce283780-7978-11ea-863a-ed982648aa22.gif)

**Comment by demisjohn** [comment]
Alternatively, the main reason to turn on interactive plotting is due to the [plots] tab's inability to Zoom/pan etc.

If the plot window operated more like a "docked" interactive Figure window (retaining zoom/pan buttons, at least for the threads still running), then one may not need to display Interactive figures in a separate window at all.
